Include/object.h

```diff
@@ -228,6 +228,23 @@ struct _object {
 /* Cast argument to PyObject* type. */
 #define _PyObject_CAST(op) _Py_CAST(PyObject*, (op))
 
+typedef union {
+    PyObject *obj;
+    uintptr_t bits;
+} _Py_TaggedObject;
+
+#define Py_OBJECT_TAG (0b0)
+
+#ifdef Py_GIL_DISABLED
+#define Py_CLEAR_TAG(tagged) ((PyObject *)((tagged).bits & ~(Py_OBJECT_TAG)))
+#else
+#define Py_CLEAR_TAG(tagged) ((PyObject *)(uintptr_t)((tagged).bits))
+#endif
+
+#define Py_OBJ_PACK(obj) ((_Py_TaggedObject){.bits = (uintptr_t)(obj)})
+
+#define Py_TAG_CAST(o) ((_Py_TaggedObject){.obj = (o)})
+
```

Comment:
We should not make any of these macros or data structures part of the public C API.